SerialPrograms/Source/NintendoSwitch/Programs/NintendoSwitch_GameEntry.cpp

```diff
@@ -312,6 +312,206 @@ bool openedgame_to_gamemenu(
+            pbf_move_joystick(context, 255, 128, 160ms, 0ms);
+        }
+        context.wait_for_all_requests();
+    }
+
+    pbf_press_button(context, BUTTON_A, 160ms, 840ms);
+
+    while (true){
+        HomeWatcher home(std::chrono::milliseconds(2000));
+        StartGameUserSelectWatcher user_select(COLOR_GREEN);
+        UpdateMenuWatcher update_menu(COLOR_PURPLE);
+        CheckOnlineWatcher check_online(COLOR_CYAN);
+        BlackScreenWatcher black_screen(COLOR_BLUE);
+        context.wait_for_all_requests();
+        int ret = wait_until(
+            stream, context,
+            std::chrono::seconds(30),
+            {
+                home,
+                user_select,
+                update_menu,
+                check_online,
+                black_screen,
+            }
+        );
+
+        //  Wait for screen to stabilize.
+        context.wait_for(std::chrono::milliseconds(100));
+
+        switch (ret){
+        case 0:
+            stream.log("Detected home screen (again).", COLOR_RED);
+            pbf_press_button(context, BUTTON_A, 160ms, 840ms);
+            break;
+        case 1:
+            stream.log("Detected user-select screen.");
+            move_to_user(context, user_slot);
+            pbf_press_button(context, BUTTON_A, 80ms, start_game_wait);
+            break;
+        case 2:
+            stream.log("Detected update menu.", COLOR_RED);
+            pbf_move_joystick(context, 128, 0, 50ms, 0ms);
+            pbf_press_button(context, BUTTON_A, 160ms, 840ms);
+            break;
+        case 3:
+            stream.log("Detected check online.", COLOR_RED);
+            context.wait_for(std::chrono::seconds(1));
+            break;
+        case 4:
+            stream.log("Detected black screen. Game started...");
+            return;
+        default:
+            OperationFailedException::fire(
+                ErrorReport::SEND_ERROR_REPORT,
+                "start_game_from_home_with_inference(): No recognizable state after 30 seconds.",
+                stream
+            );
+        }
+    }
+}
+
+void start_game_from_home(
+    VideoStream& stream, JoyconContext& context,
+    bool tolerate_update_menu,
+    uint8_t game_slot,
+    uint8_t user_slot,
+    Milliseconds start_game_mash
+){
+    context.wait_for_all_requests();
+    if (stream.video().snapshot()){
+        stream.log("start_game_from_home(): Video capture available. Using inference...");
+        start_game_from_home_with_inference(stream, context, game_slot, user_slot, start_game_mash);
+        return;
+    }else{
+        stream.log("start_game_from_home(): Video capture not available.", COLOR_RED);
+    }
+
+    if (game_slot != 0){
+        ssf_press_button(context, BUTTON_HOME, ConsoleSettings::instance().SETTINGS_TO_HOME_DELAY0, 160ms);
+        for (uint8_t c = 1; c < game_slot; c++){
+            pbf_move_joystick(context, 255, 128, 80ms, 0ms);
+        }
+    }
+
+    if (tolerate_update_menu){

```

Comment:
I think we can simplify this logic now and assume that feedback is available. This fallback system was for the old SwSh programs that could run without feedback.